src/chat.rs

```diff
@@ -3906,7 +3904,6 @@ mod tests {
     async fn test_self_talk() -> Result<()> {
         let t = TestContext::new_alice().await;
         let chat = &t.get_self_chat().await;
-        assert_eq!(DC_CONTACT_ID_SELF, ContactId::new(1));
```

Comment:
why remove that check? it makes sure, the constant is not modified accidentally and the correct values go to the database (we're testing lots of constants against their integer values and i think, that makes some sense)

Reply:
I guess because `chat.rs` should not "know" that this constant is 1. Of course we could generally re-add this test to `contact.rs`.

Reply:
OTOH it's even part of the public interface (e.g. DC Android "knows" that this constant is 1) - anyway I think that I don't have an opinion on this.

Reply:
> OTOH it's even part of the public interface 

yip, that is the idea of the check. the value of the constant does matter much to the outside -  not checking it somewhere inside, all tests will pass if the value is set to `2` for whatever reason.

Reply:
If you follow this line through the transforms it looked like this:
```
assert_eq!(DC_CONTACT_ID_SELF, ContactId::new(1));
assert_eq!(ContactId::SELF, ContactId::new(1));
assert_eq!(ContactId::SELF, ContactId::SELF);
// well fine, guess i'll delete this
```
At which point I indeed did consider if there's any point in writing a test to make sure the numeric value is 1 or not.  This would probably have been a test for `assert_eq!(ContactId::SELF.0, 1);` in `contact.rs`, but really now you're just typing out the 1 twice.  So I decided to just leave it.

> OTOH it's even part of the public interface (e.g. DC Android "knows" that this constant is 1)

This I would disagree with, DC Android does -at least theoretically- not know this.  It only knows the FFI interface which says a contact ID is a `u32` and this specific one is equal to `DC_CONTACT_ID_SELF`, but it is up to the core to decide what value `DC_CONTACT_ID_SELF` is, it can even change it with every release if it wants (no, this is a terribly bad idea, I did say "theoretically").

Reply:
In Java you can't use constants from a `.h` file, so constants to be defined in Java again: https://github.com/deltachat/deltachat-android/blob/master/src/com/b44t/messenger/DcContact.java#L5

Reply:
> At which point I indeed did consider if there's any point in writing a test to make sure the numeric value is 1 or not.  This would probably have been a test for `assert_eq!(ContactId::SELF.0, 1);` in `contact.rs`, but really now you're just typing out the 1 twice. So I decided to just leave it.

yip, that would be totally fine. i just want to make sure that there is a check at some point.

Reply:
> In Java you can't use constants from a `.h` file, so constants to be defined in Java again: https://github.com/deltachat/deltachat-android/blob/master/src/com/b44t/messenger/DcContact.java#L5

Whoa!  I assumed the JNI would be taking care of this and exporting it as a constants.  That's crazy, but thanks for pointing it out.

Reply:
> > At which point I indeed did consider if there's any point in writing a test to make sure the numeric value is 1 or not.  This would probably have been a test for `assert_eq!(ContactId::SELF.0, 1);` in `contact.rs`, but really now you're just typing out the 1 twice. So I decided to just leave it.
> 
> yip, that would be totally fine. i just want to make sure that there is a check _somewhere_ left.

fair enough, I've added this back as a test since you'd prefer